src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputExtension.java

```diff
@@ -0,0 +1,20 @@
+package org.jenkinsci.plugins.workflow.support.steps.input;
+
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+
+public interface InputExtension extends ExtensionPoint {
+    /**
+     * Callback this when an inputStep is executing.
+     * Just for notification, will ignore all exception and log them.
+     * @param inputStep instance of InputStep
+     * @param run Current building run instance
+     */
+    void input(InputStep inputStep, Run run);
```

Comment:
`notifyInput` would be the customary naming scheme for something like this, I think.